neuron/src/app/Neuron/CLI/Open.hs

```diff
@@ -0,0 +1,23 @@
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE RecordWildCards   #-}
+
+module Neuron.CLI.Open
+  ( openLocallyGeneratedFile,
+  )
+where
+
+import qualified Data.Text as T
+import Development.Shake (Action)
+import Neuron.CLI.Types (OpenCommand (..))
+import Relude
+import Rib.Shake (ribOutputDir)
+import System.FilePath
+import System.Info (os)
+import System.Posix.Process
+
+openLocallyGeneratedFile :: OpenCommand -> Action ()
+openLocallyGeneratedFile OpenCommand {..} = do
+  let htmlId = maybe "index" T.unpack zettelId
```

Comment:
This seems to break `neuron open`. It should open `z-index.html`. But this PR makes it open `index.html` (which may not even exist).

Also you should construct a route, and then use `routeUrl`.

Do something like this:

```haskell
htmlFile = Rib.routeUrl $ maybe Route_ZIndex Route_Zettel zettelId
```

Reply:
Opening `index.html` in case of no arguments is to preserve backward compatibility as this is what's happening right now: https://github.com/srid/neuron/blob/2d7c43510021d3cdeda8aed3468ba17a368b5140/neuron/src/app/Neuron/CLI.hs#L65 it's not opening `z-index.html`.

Reply:
Ah, I see. That's buggy code in `master`. If index.md doesn't exist (not all zettelkastens have them), `neuron open` will break. We should just make it open the z-index (which always exists) by default. Users can do `neuron open --id index` to open the index zettel explicitly.

Reply:
Alternatively, we could make it open index.html *if* the index zettel exists (open zindex otherwise). In which case, we will have to add a `--zindex` option to the CLI spec below, as a way to explicitly open the z-index even if index.md exists.

Reply:
You can get the index zettel like this: https://github.com/srid/neuron/blob/cf1dfa8b28ad6e9d011d46015bfd80f1fb6c2552/neuron/src/app/Neuron/Web/View.hs#L103